engine/src/main/java/org/hibernate/validator/internal/cfg/context/CascadableConstraintMappingContextImplBase.java

```diff
@@ -160,19 +160,15 @@ public boolean isCascading() {
 	}
 
 	protected CascadingMetaDataBuilder getCascadingMetaDataBuilder() {
-		Map<TypeVariable<?>, CascadingMetaDataBuilder> typeParametersCascadingMetaData = containerElementContexts.values().stream()
-				.filter( c -> c.getContainerElementCascadingMetaDataBuilder() != null )
-				.collect( Collectors.toMap( c -> c.getContainerElementCascadingMetaDataBuilder().getTypeParameter(),
-						c -> c.getContainerElementCascadingMetaDataBuilder() ) );
-
-		for ( ContainerElementConstraintMappingContextImpl typeArgumentContext : containerElementContexts.values() ) {
-			CascadingMetaDataBuilder cascadingMetaDataBuilder = typeArgumentContext.getContainerElementCascadingMetaDataBuilder();
-			if ( cascadingMetaDataBuilder != null ) {
-				typeParametersCascadingMetaData.put( cascadingMetaDataBuilder.getTypeParameter(), cascadingMetaDataBuilder );
-			}
-		}
-
-		return CascadingMetaDataBuilder.annotatedObject( configuredType, isCascading, typeParametersCascadingMetaData, groupConversions );
+		return CascadingMetaDataBuilder.annotatedObject(
+				configuredType,
+				isCascading,
+				containerElementContexts.values().stream()
+						.map( ContainerElementConstraintMappingContextImpl::getContainerElementCascadingMetaDataBuilder )
+						.filter( Objects::nonNull )
+						.collect( Collectors.toList() ),
```

Comment:
I think I would prefer to keep a local variable for debugging.

BTW, I'm not convinced of the values of using streams + method references here. And as we're trying to reduce them, we could simply remove it.

Another thing is I wonder if we should try to merge the changes about cascading early to get them out of the way?

It might not be totally easy to isolate them but I think it would have value.